### PR TITLE
Feature/setup flask

### PR DESCRIPTION
**Pull request by xal9wiii4ik**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+logging.basicConfig(format='%(asctime)s - %(levelname)s - %(message)s', level=logging.INFO)
+
+app = Flask(__name__, instance_relative_config=True)
+app.config.from_mapping(
+    SECRET_KEY=settings.SECRET_KEY
+)
+
+
+@app.route('/')
+def index() -> tp.Tuple[Response, int]:
```
Comment (artsiom-streetshares) [inline app/main.py]: lets add a basic test for this route

File: app/main.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+from flask import Flask, jsonify, Response
+
+import settings
+
+logging.basicConfig(format='%(asctime)s - %(levelname)s - %(message)s', level=logging.INFO)
+
+app = Flask(__name__, instance_relative_config=True)
+app.config.from_mapping(
+    SECRET_KEY=settings.SECRET_KEY
```
Comment (artsiom-streetshares) [inline app/main.py]: you are not feeding you config from settings.py here

File: app/main.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import typing as tp
+import logging
+
+from flask import Flask, jsonify, Response
+
+import settings
+
+logging.basicConfig(format='%(asctime)s - %(levelname)s - %(message)s', level=logging.INFO)
```
Comment (artsiom-streetshares) [inline app/main.py]: this is fine for now, later it's better to use dict/file config

File: app/main.py (hunk 2 above)
Comment (xal9wiii4ik) [reply app/main.py]: can you explain?

File: .github/workflows/deploy.yml [hunk 4]
```diff
@@ -1,49 +1,52 @@
...
-      - name: download poetry
+      - uses: actions/checkout@v2
+      - name: Create File app/.env
         run: |
-          curl https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py > get-poetry
-      - id: Install_poetry
-        run: python get-poetry
-
-      - name: activate poetry
+          echo SECRET_KEY='146EtHWvxaInR3YB3d2Cdtn4emQcUBWw' > app/.env
```
Comment (artsiom-streetshares) [inline .github/workflows/deploy.yml]: this is fine for now. generally it's bad to store the passwords and other secrets plain text in the source code. you'll need to use some sort of secret storage/encryption

File: .github/workflows/deploy.yml [hunk 5]
```diff
@@ -1,49 +1,52 @@
...
+      - name: Show file app/env content
+        run: |
+          pwd
+          ls -la
+          cat app/.env
+      - name: Create File app/env.db
         run: |
-          export PATH = "${PATH}:/root/.poetry/bin"
-      - name: install dependencies
+          echo POSTGRES_USER=flask_user > app/.env.db
```
Comment (artsiom-streetshares) [inline .github/workflows/deploy.yml]: why you need SQL_USER, SQL_PASSWORD etc and POSTGRES_USER etc? aren't those the same?

File: .github/workflows/deploy.yml [hunk 6]
```diff
@@ -1,49 +1,52 @@
...
+          pwd
+          ls -la
+          cat app/.env.db
+      - name: Build the stack
+        run: docker-compose up -d
+      - name: Test
+        env:
+          COMPOSE_INTERACTIVE_NO_CLI: 1
+        run: docker-compose exec -T web pytest
+      - name: PreCommit install
```
Comment (artsiom-streetshares) [inline .github/workflows/deploy.yml]: use separate GH actions for test and pre-commit, so they can run in parallel

File: .gitignore [hunk 7]
```diff
@@ -150,4 +150,5 @@ cython_debug/
 #  and can be added to the global gitignore or merged into this file.  For a more nuclear
 #  option (not recommended) you can uncomment the following to ignore the entire idea folder.
 .idea/
-app/.env.db
+
+/app/.env.db
```
Comment (artsiom-streetshares) [inline .gitignore]: hmm, slash at the beginning means the file is relative to the system root directory. here you should specify things relative to your project root

File: pyproject.toml [hunk 8]
```diff
@@ -10,6 +10,7 @@ requests = "^2.26.0"
 Flask = "^2.0.2"
 python-dotenv = "^0.19.2"
 Flask-SQLAlchemy = "^2.5.1"
+pytest = "^6.2.5"
```
Comment (artsiom-streetshares) [inline pyproject.toml]: this is a dev dependency as well

**Comment by artsiom-streetshares** [comment]
I see it's mergeable without my approve. it shouldn't be the case

File: .github/workflows/deploy.yml (hunk 5 above)
Comment (artsiom-streetshares) [reply .github/workflows/deploy.yml]: also the workflow name is `deploy`, but it should be `test` or similar

**Comment by artsiom-streetshares** [comment]
also, why the branch is setup flask? you should ideally make branch names dedicated to the feature you develop and don't mix other features in there. also provide meaningful description to all of your PRs. Here it could be "Add a github action for running tests and pre-commit checks"

File: .github/workflows/deploy.yml [hunk 9]
```diff
@@ -1,49 +1,52 @@
...
+        run: docker-compose exec -T web poetry add pylint mypy types-requests
```
Comment (artsiom-streetshares) [inline .github/workflows/deploy.yml]: why you need pylint? I though you use flake8 in your pre-commit

File: .github/workflows/deploy.yml (hunk 4 above)
Comment (artsiom-streetshares) [reply .github/workflows/deploy.yml]: https://www.theserverside.com/blog/Coffee-Talk-Java-News-Stories-and-Opinions/GitHub-Actions-Secrets-Example-Token-Tutorial as an option something like this. but again, this is fine for now.

File: app/main.py (hunk 2 above)
Comment (artsiom-streetshares) [reply app/main.py]: I see you've removed the file. I mean't you didn't use your settings here, but used some hardcoded values.

File: .github/workflows/deploy.yml [hunk 10]
```diff
@@ -1,49 +1,33 @@
-# This is a basic workflow to help you get started with Actions
-
 name: CI
```
Comment (artsiom-streetshares) [inline .github/workflows/deploy.yml]: rename that to `test`

File: .github/workflows/pre-commit.yml [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Create File app/.env
+        run: |
+          echo SECRET_KEY='some_secret' > app/.env
+
+          echo DATABASE='postgresql' >> app/.env
+          echo SQL_HOST='db' >> app/.env
```
Comment (artsiom-streetshares) [inline .github/workflows/pre-commit.yml]: user either `SQL_` or `POSTGRES_` prefix

File: pyproject.toml [hunk 12]
```diff
@@ -12,12 +12,14 @@ python-dotenv = "^0.19.2"
 Flask-SQLAlchemy = "^2.5.1"
 psycopg2-binary = "^2.9.2"
 gunicorn = "^20.1.0"
+pre-commit = "^2.16.0"
```
Comment (artsiom-streetshares) [inline pyproject.toml:15]: this is a dev dependency

File: .github/workflows/deploy.yml (hunk 10 above)
Comment (xal9wiii4ik) [reply .github/workflows/deploy.yml]: fixed

File: app/main.py (hunk 1 above)
Comment (xal9wiii4ik) [reply app/main.py]: done

File: app/main.py (hunk 2 above)
Comment (xal9wiii4ik) [reply app/main.py]: done

File: app/main.py (hunk 3 above)
Comment (xal9wiii4ik) [reply app/main.py]: will done in future(this is fine for now)

File: .github/workflows/deploy.yml (hunk 4 above)
Comment (xal9wiii4ik) [reply .github/workflows/deploy.yml]: done(test values)

File: .github/workflows/deploy.yml (hunk 5 above)
Comment (xal9wiii4ik) [reply .github/workflows/deploy.yml]: done

File: .github/workflows/deploy.yml (hunk 6 above)
Comment (xal9wiii4ik) [reply .github/workflows/deploy.yml]: done

File: .gitignore (hunk 7 above)
Comment (xal9wiii4ik) [reply .gitignore]: done

File: pyproject.toml (hunk 8 above)
Comment (xal9wiii4ik) [reply pyproject.toml]: done

File: .github/workflows/deploy.yml (hunk 9 above)
Comment (xal9wiii4ik) [reply .github/workflows/deploy.yml]: done

File: .github/workflows/pre-commit.yml (hunk 11 above)
Comment (xal9wiii4ik) [reply .github/workflows/pre-commit.yml]: done

File: pyproject.toml (hunk 12 above)
Comment (xal9wiii4ik) [reply pyproject.toml:15]: done

**Pull request merged**